src/soter/boringssl/soter_asym_cipher.c

```diff
@@ -121,7 +121,7 @@ soter_status_t soter_asym_cipher_encrypt(soter_asym_cipher_t* asym_cipher, const
 {
 	EVP_PKEY *pkey;
 	RSA *rsa;
-	int rsa_mod_size;
+	size_t rsa_mod_size;
```

Comment:
not sure about this change.
we should double check no compatibility issues arise (on x32 - x64).
i briefly read through code, and this change might affect size calculations on 150 on x64
https://github.com/cossacklabs/themis/pull/377/files#diff-b4bb5a3099e56a8faec5df05d0836579R150

Reply:
Got it 👍  I'll review the `size_t` casts and provide more substantial justification for why exactly they are safe to do.

Reply:
The changes in this PR should be okay (see below).

However, `git grep` reveals thirty-something other places where we use `(size_t)` casts. They are mostly casts from OpenSSL types involving comparisons. To get it absolutely right I will have to review variable usage as well to ensure that we don't _store_ size_t values in data structures and buffers. I guess that might a different story.

#### jni/themis_session.c

`GetArrayLength()` returns a `jsize` which is defined as `jint` (32-bit signed integer). `jsize` is intended for indices and sizes and should be non-negative. However, JVM does not have a concept of unsigned integers per se. It should be safe to cast `jsize` as `size_t` according to 6.3.1.3 of C99 standard. (Caveat: on 32-bit+ platforms, it will break on 16-bit platforms but who uses them now...)

This value is only used for comparison and never stored to other data structures.

#### src/soter/boringssl/soter_asym_cipher.c

`rsa_mod_size` is initialized by `RSA_size()` which returns `int` (due to OpenSSL API). It returns the size of RSA modulus in bytes which should be non-negative. Casting is safe (provided that int is not bigger than size_t). The value is then compared to `size_t plain_data_length` after some arithmetics. It is never stored to memory.

#### src/soter/{boringssl,openssl}/soter_sym.c

`size_t iv_length` is compared to the result of `EVP_CIPHER_iv_length()`. In newer versions this is a function that returns `unsigned int`, but in older versions this is a macro that refers to `int iv_len` field of struct EVP_CIPHER. IV length in bytes should be a non-negative value. Casting it as `size_t` should be safe. The value is used only for comparisons.

#### tests/themis/themis_secure_session.c

Here `ssize_t bytes_sent` is compared to `size_t current_length`. It is safe to cast it here because previously we check that `bytes_sent == bytes_received` and before that `bytes_received > 0`. So again it should be non-negative and okay to cast. And we never store it anywhere.

Reply:
thank you for explanations! I reviewed each item, and thought about possible incompatibilities, but haven't found any valid examples.